apps/prairielearn/src/lib/group-update.js

```diff
@@ -1,337 +1,232 @@
-                updateList.push([groupName, uid]);
-              })
-              .then(
-                () => {
-                  let params = [assessment_id, updateList, authn_user_id];
-                  sqldb.call('assessment_groups_update', params, (err, result) => {
-                    const allCount = updateList.length;
-                    let successCount = 0,
-                      errorCount = 0;
-                    if (err) {
-                      job.verbose(String(err)); //server error
-                      errorCount = allCount; //all failed
-                    } else {
-                      const notExist = result.rows[0].not_exist_user;
-                      const inGroup = result.rows[0].already_in_group;
-                      if (notExist) {
-                        job.verbose(`----------------------------------------`);
-                        job.verbose(
-                          `ERROR: The following users do not exist. Please check their uids first.`,
-                        );
-                        notExist.forEach((user) => {
-                          job.verbose(user);
-                        });
-                        errorCount += notExist.length;
-                      }
-                      if (inGroup) {
-                        job.verbose(`----------------------------------------`);
-                        job.verbose(`ERROR: The following users are already in a group.`);
-                        inGroup.forEach((user) => {
-                          job.verbose(user);
-                        });
-                        errorCount += inGroup.length;
-                      }
-                      successCount = allCount - errorCount;
-                    }
-                    job.verbose(`----------------------------------------`);
-                    namedLocks.releaseLock(lock, (lockErr) => {
-                      if (lockErr) {
-                        job.fail(`ERROR: The lock ${lockName} was not released successfully`);
-                        return;
-                      }
-                    });
-                    job.verbose(`Released lock ${lockName}`);
-                    if (errorCount === 0) {
-                      job.verbose(
-                        `Successfully updated groups for ${successCount} students, with no errors`,
-                      );
-                      job.succeed();
-                    } else {
-                      job.verbose(`Successfully updated groups for ${successCount} students`);
-                      job.fail(`Error updating ${errorCount} students`);
-                    }
-                  });
-                },
-                (err) => {
-                  job.fail(error.newMessage(err, 'Error processing CSV'));
-                },
-              );
-          }
+
+  const result = await sqldb.queryOneRowAsync(sql.select_assessment_info, { assessment_id });
```

Comment:
I don't know if you're avoiding it for a reason, but this (and the similar query in the other function) could also have `queryRow`.

Reply:
I think my original plan was to minimize the number of other changes in this PR. You're right, these are easy to convert, I'll do that now.